### DISCUSSION TITLE
采集过程中暂停采集器，checkpoint中保存的文件offset不正确

### DISCUSSION THREAD

**Discussion by samtangweicheng** [alibaba/ilogtail#1657] [Help]
我们基于flusher_opentelemetry做了二次开发用于上报数据。
新的flusher_xxx_otlp的flush方法中会读取各个logRecord的"__file_offset__"标签，然后打印出来。
![image](https://github.com/user-attachments/assets/810abd97-1971-4eb3-8244-f600715f90e7)

在测试的时候，我们发现重启采集器后有漏采的情况：
1、采集配置：
`{
  "enable" : true,
  "global" : {
    "EnableTimestampNanosecond" : true
  },
  "inputs" : [ {
    "Type" : "input_file",
    "FilePaths" : [ "/root/zhl/log_file1.log4" ],
    "MaxDirSearchDepth" : 5,
    "ExcludeFilePaths" : [ ],
    "TailSizeKB" : 10485760,
    "AppendingLogPositionMeta" : true,
    "AllowingIncludedByMultiConfigs" : true
  } ],
  "flushers" : [ {
    "Type" : "flusher_xxx_otlp",
    "Logs" : {
      "Endpoint" : "xxxxxx.xxxxxx.cn:12345",
      "Timeout" : 10000,
      "WaitForReady" : true,
      "Compression" : "gzip"
    }
  } ]
}
`
2、启动采集器后，用touch命令更新目标文件的修改时间，让采集器开始采集。/root/xxx/log_file1.log2是一个20M大小左右的文件。
3、1-2秒后给采集器发送sigTrem信号停止采集器；
4、重新运行采集器，然后在用touch命令更新目标文件的修改时间，让采集器继续采集。

在第三步之后，打开ilogtail.LOG日志文件如下，能看到是从0开始采集器的：
![image](https://github.com/user-attachments/assets/195d3c3a-fdc2-493e-b9a9-8edae5b9e843)

logtail_plugin.LOG日志中也打印出了最后一次发送的数据包的最大offset是5229336
![image](https://github.com/user-attachments/assets/f896931b-354e-4e34-83a9-7696cd1e583a)

但是此时查看checkpoint文件，发现其中的offset是8388480
![企业微信截图_17225013855869](https://github.com/user-attachments/assets/72ebd417-bdc7-4a0b-8614-fa89e90481a5)

重启后，ilogtail.LOG中显示是从8388480处继续采集
![企业微信截图_17225014931361](https://github.com/user-attachments/assets/adf5e9f8-d034-4a01-ace5-2a0bb50605ca)

中间的5229336到8388480 这段日志漏采了。

**Comment by Takuka0311** [comment]
正常的流程，在第三步停止时，logtail会自动dump checkpoint到本地，并在重启后自动读取checkpoint继续采集，不需要再用touch命令更新目标文件的修改时间。如果有完整的重启前后的日志可能更容易分析
